### ISSUE TITLE
Remove all incorrect role designations

### ISSUE THREAD

**Issue by jechols** [open-oni/open-oni#315]
Most of the time we shouldn't specify `role="..."` anywhere.  Unless we've actually built a widget and ensured that it conforms to the requirements of whatever role we give it (which is often very difficult to do by hand).

In particular, right now we definitely have a bunch of places we claim `<a>` tags are buttons.  Without the JavaScript to make those tags *behave* as buttons, however, we have a WCAG failure.  It's also possible that our styling will become a failure in the future, because we make those links *look* like buttons, which can be a problem for certain cognitive impairments.

More reading: http://www.webaxe.org/proper-use-buttons-links/ (ironically the font on that site is terrible for poor vision....)
